embeddings/hyperparameter_search/configspace.py

```diff
@@ -102,3 +103,28 @@ def _retrieve_embedding_type(embedding_name: str) -> str:
         else:
             raise TypeError("Embedding type not recognized")
         return embedding_type
+
+    @staticmethod
+    def _parse_yaml_params(parameters: Dict[str, Any]) -> Dict[str, Parameter]:
+        parsed_parameters = {}
+        for param_key, param_values in parameters.items():
+            param_type = param_values.pop("param_type")
+            param: Parameter
+            if param_type == "constant":
+                param = ConstantParameter(**param_values)
+            elif param_type == "searchable":
+                param = SearchableParameter(**param_values)
+            else:
+                raise ValueError("Unrecognized parameter type")
```

Comment:
maybe we should not raise error and just log it?
import logging
logger = logging.getLogger(__name__)
logger.error(... exc_info=True)

Reply:
According to the convention in this project, raising the error should be sufficient. Moreover, I am not pleased when I handle some error in `try catch`, but still error is logged :p

Reply:
Yep - logging error usually means you log handling it, nonetheless your ValueError would make more sense if you specify what casued it not with generic messege so that someone can search for string causing error (providing more info / context / clearer cause" like: Unrecognized parameter type: param_type"

Reply:
added f-string with param_type